packages/flutter/lib/src/foundation/_identity_map_io.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:collection';
+
+/// Create a new identity [Map].
+///
+/// This cannot be used with classes that override [Object.hashCode] or
+/// [Object.==].
+///
+/// On the web this will be backed by a JavaScript Map, which is more
+/// performant if the type of `K` is not a [String], [int], [double], or [bool].
+///
+/// On Mobile and desktop this returns an identity [HashMap].
+Map<K, V> createIdentityMap<K, V>() {
+  // On mobile/desktop type objects are not necessarily canonicalized in the same
+  // way as on the web. Use a regular map for these.
+  if (K == Type) {
+    return HashMap<K, V>();
+  }
+  return HashMap<K, V>.identity();
+}
```

Comment:
Shouldn't this be an extension of HashMap or at least somehow replace `identity`? It seems it is a better way, but no one will use because it is hard to discover.

Reply:
For the general es6 map issue see https://github.com/dart-lang/sdk/issues/45092. That would need to be implemented in the Dart SDK. The goal of this PR is only to measure if there is a performance improvement for Flutter.